src/Aspire.Hosting.Azure.AppContainers/AzureContainerAppsInfrastructure.cs

```diff
@@ -886,8 +885,7 @@ private void AddContainerRegistryParameters(ContainerAppConfiguration app)
     // This is a workaround to handle nested formattable strings until the bug is fixed.
     private static BicepValue<string> Interpolate(BicepValueFormattableString text)
     {
-        var formatStringBuilder = new StringBuilder();
-        var arguments = new List<BicepValue<string>>();
+        var bicepStringBuilder = new BicepStringBuilder();
```

Comment:
FYI - @davidfowl. We aren't able to get rid of our `Format` string parsing code from ReferenceExpression. But we are able to use the new `BicepStringBuilder` here.

Reply:
I think I could get rid of the BicepValueFormattableString class and just pass the `ReferenceExpression` around. No need in creating a new FormattableString object.

Reply:
We should be able to if you move this logic into the caller right?

Reply:
Yes, I can build up the BicepStringBuilder earlier and not have the BicepValueFormattableString at all.

Reply:
Ah yea, ok.